src/TypesFinder/FindTypeFromAst.php

```diff
@@ -19,10 +19,16 @@ class FindTypeFromAst
      */
     public function __invoke($astType, LocatedSource $locatedSource, $namespace = '')
     {
-        $context = (new ContextFactory())->createForNamespace(
-            $namespace,
-            $locatedSource->getSource()
-        );
+        static $contexts = [];
```

Comment:
@asgrim @Ocramius any ideas for improving this? could this be made into a statefull service?